src/common/pool_map.c

```diff
@@ -1590,8 +1603,12 @@ gen_pool_buf(struct pool_map *map, struct pool_buf **map_buf_out, int map_versio
 
 	rc = add_domain_tree_to_pool_buf(map, map_buf, map_version, dss_tgt_nr, ndomains,
 					 domains, ordered_ranks);
-	if (rc != 0)
+	if (rc != 0) {
+		/* Do not need update the pool map anymore */
+		if (rc == -DER_EXIST)
+			rc = 0;
 		D_GOTO(out_map_buf, rc);
+	}
```

Comment:
[Not requesting changes] I was confused during the review that a function named `gen_pool_buf` may return 0 while not generating a map buffer. One could define a positive return value for `gen_pool_buf` to indicate the -DER_EXIST case here, for example, if he is uncomfortable with a common negative return value. Or, at least a comment for `gen_pool_buf` should be added to warn innocent users.

Reply:
yeah, indeed, export DER_EXIST to the caller might make thing clearer.  I will update it in the following patch, if it is only for better understanding. Thanks.